lib/exercism/github.rb

```diff
@@ -1,40 +1,47 @@
 class Github
+  include Connection
+  class UnauthorizedUser < StandardError; end
+  USER_AGENT = 'github.com:exercism/exercism.io'
+
   def self.login_url(q)
     "https://github.com/login/oauth/authorize?#{q.map { |k, v| '%s=%s' % [k, v] }.join('&')}"
   end
 
-  def self.connect_to(url)
-    Faraday.new(url: url) do |c|
-      c.use Faraday::Response::Logger
-      c.use Faraday::Adapter::NetHttp
+  def self.user_info(access_token)
+    conn = connect_to 'https://api.github.com'
+    response = conn.get do |req|
+      req.url '/user'
+      req.headers['User-Agent'] = USER_AGENT
+      req.params['access_token'] = access_token
     end
+    raise UnauthorizedUser unless response.success?
+    result = JSON.parse(response.body)
+    GithubUser.new(result['id'], result['login'], result['email'], result['avatar_url'])
```

Comment:
An alternate signature that would allow you to pass in the `result` hash and eliminate another set of 4 parameters:

```ruby
GithubUser.new(result)

...

class GithubUser
  def initialize(attributes)
    @id         = attributes.fetch('id')
    @username   = attributes.fetch('login')
    @email      = attributes.fetch('email')
    @avatar_url = attributes.fetch('avatar_url')
  end
end